submissions/NastjonkaK/OOP_exercise/index.js

```diff
@@ -0,0 +1,84 @@
+class Creature {
+  constructor(name, saying) {
+    this.name = name,
+    this.saying = saying
+  }
+}
+
+class Animals extends Creature {
+  constructor(name, gender, saying) {
+    super(name, saying);
+    this.gender = gender,
+    this.paws = 4
+  }
+
+  greeting() {
+    return `\ Hello! I am a ${this.type}, ${this.gender}. My name is ${this.name}. I have ${this.paws} paws and a tail. And I like to say: "${this.saying}".`;
+  }
+}
+
+class Human extends Creature {
+  constructor(name, saying) {
+    super(name, saying)
+    this.type = "human",
+    this.legs = 2,
+    this.hands = 2
+  }
+
+  greeting() {
+    return `\ Hello! I am a ${this.type}, ${this.gender}. My name is ${this.name}. I have ${this.legs} legs and ${this.hands} hands. And I want to say: "${this.saying}".`;
+  }
+}
+
+class Man extends Human {
+  constructor(name, saying) {
+    super(name, saying);
+    this.type = 'man',
+    this.gender = 'male'
+  }
+}
+
+class Woman extends Human {
+  constructor(name, saying) {
+    super(name, saying);
+    this.type = 'woman',
+    this.gender = 'female'
+  }
+}
+
+
+class Dog extends Animals {
+  constructor(name, gender) {
+    super(name, gender);
+    this.type = "dog",
+    this.saying = "woof-woof"
+  }
+}
+
+class Cat extends Animals {
+  constructor(name, gender) {
+    super(name, gender);
+    this.type = "cat",
+    this.saying = "meow-meow"
+  }
+}
+
+class HumanCat extends Woman {
```

Comment:
```
cat-woman's saying should be exactly the same as cat's
whenever you change cat's saying cat-woman's saying should change accordingly, they are strongly tied on some astral level
```
[Task](https://github.com/OleksiyRudenko/a-tiny-JS-world#the-job)

I think you shoud rewrite your class HumanCat. Pass `cat.saying` as a parameter is a good idea but not in this case.